ecs-cli/modules/cli/compose/project/project_test.go

```diff
@@ -318,3 +325,65 @@ func setupTestProjectWithEcsParams(t *testing.T, ecsParamsFileName string) *ecsP
 		ecsContext: ecsContext,
 	}
 }
+
+func TestParsePrivateRegistryContext(t *testing.T) {
+	credsInputString := `version: "1"
+registry_credential_outputs:
+  task_execution_role: someTestRole
+  container_credentials:
+    my.example.registry.net:
+      credential_parameter: arn:aws:secretsmanager::secret:amazon-ecs-cli-setup-my.example.registry.net
+      container_names:
+      - web
+    another.example.io:
+      credential_parameter: arn:aws:secretsmanager::secret:amazon-ecs-cli-setup-another.example.io
+      kms_key_id: arn:aws:kms::key/some-key-57yrt
+      container_names:
+      - test`
+
+	content := []byte(credsInputString)
+
+	tmpfile, err := ioutil.TempFile("", regcredio.ECSCredFileBaseName)
+	assert.NoError(t, err, "Could not create ecs fields tempfile")
+
+	credFileName := tmpfile.Name()
+	defer os.Remove(credFileName)
+
+	project := setupTestProjectWithPrivateRegistryContext(t, "", credFileName)
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs fields tempfile")
+
+	if err := project.parsePrivateRegistryContext(); err != nil {
+		t.Fatalf("Unexpected error parsing the "+regcredio.ECSCredFileBaseName+" data [%s]: %v", credsInputString, err)
+	}
+
+	ecsRegCreds := project.privateRegistryContext
+	assert.NotNil(t, ecsRegCreds, "Expected "+regcredio.ECSCredFileBaseName+" to be set on project")
+	assert.Equal(t, "1", ecsRegCreds.Version, "Expected Version to match")
+
+	credResources := ecsRegCreds.CredentialResources
+	assert.NotNil(t, credResources, "Expected credential resources to be non-nil")
+	assert.Equal(t, "someTestRole", credResources.TaskExecutionRole)
+	assert.NotNil(t, credResources.ContainerCredentials, "Expected ContainerCredentials to be non-nil")
+
+	firstOutputEntry := credResources.ContainerCredentials["my.example.registry.net"]
+	assert.NotEmpty(t, firstOutputEntry)
+	assert.Equal(t, "arn:aws:secretsmanager::secret:amazon-ecs-cli-setup-my.example.registry.net", firstOutputEntry.CredentialARN)
+	assert.Equal(t, "", firstOutputEntry.KMSKeyID)
+	assert.ElementsMatch(t, []string{"web"}, firstOutputEntry.ContainerNames)
+
+	secondOutputEntry := credResources.ContainerCredentials["another.example.io"]
+	assert.NotEmpty(t, secondOutputEntry)
+	assert.Equal(t, "arn:aws:secretsmanager::secret:amazon-ecs-cli-setup-another.example.io", secondOutputEntry.CredentialARN)
+	assert.Equal(t, "arn:aws:kms::key/some-key-57yrt", secondOutputEntry.KMSKeyID)
+	assert.ElementsMatch(t, []string{"test"}, secondOutputEntry.ContainerNames)
+}
+
+func TestParsePrivateRegistryContext_NoFile(t *testing.T) {
+	project := setupTestProject(t)
+	err := project.parsePrivateRegistryContext()
+	if assert.NoError(t, err) {
+		assert.Nil(t, project.privateRegistryContext)
+	}
```

Comment:
I assume there will be more tests for error cases (e.g. repeated container names in the same regcred entry, two regcreds entry associated with the same container, etc)

Reply:
yes! there is a test for an error on duplicate container names in `convert_task_definition_test.go`